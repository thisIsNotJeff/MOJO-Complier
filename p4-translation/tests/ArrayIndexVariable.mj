var x: [10]int;
def Get(n: int): int {
  return x[n];
}