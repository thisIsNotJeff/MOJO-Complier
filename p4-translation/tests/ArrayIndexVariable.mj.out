	.data
	.balign 8
ArrayIndexVariable.x:
	.quad 0
	.quad 0
	.quad 0
	.quad 0
	.quad 0
	.quad 0
	.quad 0
	.quad 0
	.quad 0
	.quad 0
	.text
ArrayIndexVariable.Get:
MOVE(
 TEMP %rax,
 MEM(
  ADD(
   NAME ArrayIndexVariable.x,
   MUL(
    TEMP ArrayIndexVariable.Get.n,
    CONST 8)), 
  CONST 0, 8)),
JUMP(
 NAME L.0),
LABEL L.0


	.text
ArrayIndexVariable:

